qutip/core/cy/qobjevo.pyx

```diff
@@ -0,0 +1,835 @@
+
+    args : dict
+        dictionary that contain the arguments for the coefficients
+
+    tlist : array_like
+        List of times corresponding to the values of the numpy-array
+        coefficients are applied.
+
+    copy : bool
+        If Q_object is already a QobjEvo, return a copy.
+
+    step_interpolation : bool
+        For array :obj:`Coefficient`, use step interpolation instead of spline.
+
+    Attributes
+    ----------
+    dims : list
+        List of dimensions keeping track of the tensor structure.
+
+    shape : (int, int)
+        List of dimensions keeping track of the tensor structure.
+
+    Property
+    --------
+    num_obj
+        Number of parts composing the system.
+
+    const:
+        Does the system change depending on `t`.
+
+    isoper:
+        Indicates if the system represents an operator.
+
+    issuper:
+        Indicates if the system represents an superoperator.
+
+    Methods
+    -------
+    copy() :
+        Create copy of Qobj
+
+    arguments(new_args):
+        Update the args of the object
+
+    Math:
+        +/- QobjEvo, Qobj, scalar:
+            Addition is possible between QobjEvo and with Qobj or scalar
+        -:
+            Negation operator
+        * Qobj, scalar:
+            Product is possible with Qobj or scalar
+        / scalar:
+            It is possible to divide by scalar only
+    conj()
+        Return the conjugate of quantum object.
+
+    dag()
+        Return the adjoint (dagger) of quantum object.
+
+    trans()
+        Return the transpose of quantum object.
+
+    &
+        tensor between Quantum Object
+
+    linear_map(op_mapping)
+        Apply the function f to every Qobj. f(Qobj) -> Qobj
+
+    tidyup(atol=1e-12)
+        Removes small elements from quantum object.
+
+    compress():
+        Merge ops which are based on the same quantum object and coeff type.
+
+    __call__(t, args={}):
+        Return the Qobj at time t.
+        args Overwrite the `args` for this call.
+
+    matmul(t, mat):
+        Product of this at t time with a Qobj.
+
+    expect(t, psi, herm=False):
+        Calculates the expectation value for the quantum object (if operator,
+            no check) and state psi.
+        Return only the real part if herm.
+
+    to(data_type):
+        Transform all component to a common data_type.
+
+    """
+    def __init__(QobjEvo self, Q_object, args=None, tlist=None,
+                 step_interpolation=False, copy=True):
+        if isinstance(Q_object, QobjEvo):
+            self.dims = Q_object.dims.copy()
+            self.shape = Q_object.shape
+            self._shift_dt = (<QobjEvo> Q_object)._shift_dt
+            self._issuper = (<QobjEvo> Q_object)._issuper
+            self._isoper = (<QobjEvo> Q_object)._isoper
+            if copy or args:
+                self.elements = (<QobjEvo> Q_object).elements.copy()
+            else:
+                self.elements = (<QobjEvo> Q_object).elements
```

Comment:
We have to copy the elements here otherwise we'll have to two QobjEvo's modifying the same list of elements resulting in confusing bugs. Perhaps there is some benefit that I'm not seeing, but I think we should remove the `copy=True` argument entirely and always copy the list of elements.

Reply:
Ah, I see we use `copy` later on to flag copying `Qobj`s. We can keep it for that, I think.

Reply:
@Ericgig Let's sort this out before merging.